### ISSUE TITLE
Use pytorch-lightning for training? 

### ISSUE THREAD

**Issue by mpariente** [mpariente/AsSteroid#14]
This would solve #6 and #3 . 
- Check flexibility with respect to : 
  - Model saving and loading (need to instantiate or not?)
  - Optimizers and callbacks
  - config.yml files in egs
  - Loss classes to handle PIT nicely
  - Unified dataset and dataloader structure? 

Any feedback on that from anybody?

**Comment by sunits** [comment]
Any estimate on the changes required?

**Comment by mpariente** [comment]
Surprisingly not much ! 
Example training scripts for the ConvTasnet recipe can be found [here](https://github.com/mpariente/AsSteroid/tree/light_ignite_trial/egs/wham/ConvTasNet).
I tried `pytorch-lightning` and `ignite`.
`ignite` is soon going to be 2 years old and `pytorch-lightning` will soon be a year old. Even so, the interface of `pytorch-lightning` is already a bit better (even if less stable and will need refactoring IMO) and more ambitious.
At this stage  I would vote for `pytorch-lightning`. 
Any thoughts  @popcornell @sunits @hbredin ?

**Comment by popcornell** [comment]
I am not familiar with either of two but I like the `pytorch-lightning` interface more.
I think is more clear. Plus its support for multi-node training seems to be better than the one in `ignite`.

**Comment by hbredin** [comment]
There is also one more candidate: https://docs.fast.ai/train.html
However, I've never used any of them.

**Comment by mpariente** [comment]
It's less easy to interface to native pytorch with `fastai`, so that's a little bit less flexible. Plus the dependencies are also quite big. On the other side, it's much more stable and very used, I'll look into it more.

**Comment by popcornell** [comment]
It seems to me that the only advantage of `fast.ai` over the other two (aside of much more wider community and better support) is that it supports some training scheduling tricks which I have to say are quite effective. 
But these can be easily ported and implemented via callbacks. 
`fastai` seems also much more focused on CV tasks so much of the features it offers will not be used.

**Comment by mpariente** [comment]
Solved by #19
